subsys/net/lib/lwm2m/lwm2m_pull_context.c

```diff
@@ -412,12 +428,14 @@ static void firmware_transfer(struct k_work *work)
 	return;
 
 error:
-	context->result_cb(ret);
-	lwm2m_engine_context_close(&context->firmware_ctx);
+	context->result_cb(context, ret);
+	cleanup_context();
 }
 
-int lwm2m_pull_context_start_transfer(struct firmware_pull_context *ctx)
+int lwm2m_pull_context_start_transfer(struct firmware_pull_context *ctx,
```

Comment:
I guess this `timeout` is the only potentially controversial part. Can you explain how it works if one attempts to start a FOTA and a swmgmt pull simultaneously?

Reply:
As I am only handling one context at a time i need a semaphore... so far so good

The "context" is being started via the system workq using the function firmware_transfer which then calls transfer_request

transfer_request then registers a download request of the binary and uses async response callbacks to handle if it is done or should ask for the next block ...

This means that transfer_request will terminate BEFORE the downaload is completed. Good in terms of no occupying the systemworkq but it also means that we need to track if the context is being used or is available with a semaphore.

So to answer your question: Which ever comes first will start downloading and the other one will have to wait until the first one is done. At which point the second one will start downloading (unless already timedout)